ctapipe/reco/tests/test_hillas_intersection.py

```diff
@@ -42,5 +52,229 @@ def test_parallel():
+
+    #  x-axis is along the altitude and y-axis is along the azimuth
+    hillas_1 = HillasParametersContainer(x=cog_coords_nom_1.delta_alt,
+                                         y=cog_coords_nom_1.delta_az,
+                                         intensity=100,
+                                         psi=0 * u.deg)
+
+    hillas_2 = HillasParametersContainer(x=cog_coords_nom_2.delta_alt,
+                                         y=cog_coords_nom_2.delta_az,
+                                         intensity=100,
+                                         psi=45 * u.deg)
+
+    hillas_3 = HillasParametersContainer(x=cog_coords_nom_3.delta_alt,
+                                         y=cog_coords_nom_3.delta_az,
+                                         intensity=100,
+                                         psi=90 * u.deg)
+
+    hillas_dict = {1: hillas_1, 2: hillas_2, 3: hillas_3}
+
+    reco_nominal = hill_inter.reconstruct_nominal(hillas_parameters=hillas_dict)
+
+    nominal_pos = SkyCoord(
+        delta_az=u.Quantity(reco_nominal[0], u.rad),
+        delta_alt=u.Quantity(reco_nominal[1], u.rad),
+        frame=nominal_frame
+    )
+
+    np.testing.assert_allclose(nominal_pos.altaz.az.to_value(u.deg), azimuth.to_value(u.deg), atol=1e-8)
+    np.testing.assert_allclose(nominal_pos.altaz.alt.to_value(u.deg), altitude.to_value(u.deg), atol=1e-8)
+
+
+def test_reconstruction():
+    """
+    a test of the complete fit procedure on one event including:
+    • tailcut cleaning
+    • hillas parametrisation
+    • direction fit
+    • position fit
+
+    in the end, proper units in the output are asserted """
+
+    filename = get_dataset_path("gamma_test_large.simtel.gz")
+
+    fit = HillasIntersection()
+
+    source = event_source(filename, max_events=10)
+
+    horizon_frame = AltAz()
+
+    for event in source:
+        array_pointing = SkyCoord(
+            az=event.mc.az,
+            alt=event.mc.alt,
+            frame=horizon_frame
+        )
+
+        hillas_dict = {}
+        telescope_pointings = {}
+
+        for tel_id in event.dl0.tels_with_data:
+
+            geom = event.inst.subarray.tel[tel_id].camera
+
+            telescope_pointings[tel_id] = SkyCoord(alt=event.mc.tel[tel_id].altitude_raw * u.rad,
+                                                   az=event.mc.tel[tel_id].azimuth_raw * u.rad,
+                                                   frame=horizon_frame)
+            pmt_signal = event.r0.tel[tel_id].waveform[0].sum(axis=1)
+
+            mask = tailcuts_clean(geom, pmt_signal,
+                                  picture_thresh=10., boundary_thresh=5.)
+            pmt_signal[mask == 0] = 0
+
+            try:
+                moments = hillas_parameters(geom, pmt_signal)
+                hillas_dict[tel_id] = moments
+            except HillasParameterizationError as e:
+                print(e)
+                continue
+
+        if len(hillas_dict) < 2:
+            continue
```

Comment:
We should make sure that at least one event was tested!
If no events survive, we will not test the hillas intersection. Please add a check that at least one, or a known number of events were indeed tested.

Reply:
I copied this from test_HillasReconstructor.py...should I change both files?

Reply:
guess so

Reply:
If it's copied, can we just have one test that loops over the reconstruction methods?

Reply:
> If it's copied, can we just have one test that loops over the reconstruction methods?

sure

Reply:
you could use this parameterize feature of pytest

Reply:
I was wondering how to do it :D thanks for the suggestion...I'm having a look at it right now

Reply:
I just pushed a new commit adding an assert over the minimum number of reconstructed events. Could you check that? If it's fine I'll just put that in a single test which tests both reconstruction methods.